crates/bevy_ecs/src/archetype.rs

```diff
@@ -658,6 +676,10 @@ impl SparseSetIndex for ArchetypeComponentId {
     }
 }
 
+/// Maps a [`ComponentId`] to the list of [`Archetypes`]([`Archetype`]) that contain it,
+/// along with the index of the column in the [`Table`](crate::storage::table::Table) where the component is stored.
+pub type ComponentIndex = HashMap<ComponentId, HashMap<ArchetypeId, ArchetypeRecord>>;
```

Comment:
If we turn the value here into a struct it could also contain the corresponding sparse set for the component. That way when you index here during a get you don't need to do another lookup.

That doesn't necessarily need to happen in this PR though.

Reply:
I don't get what you mean by "it could also contain the corresponding sparse set for the component".
The ComponentId itself it enough to get the BlobVec column from `SparseSets`, so what is the extra data that we could cache here?

Reply:
We could move the sparse set itself here so you don't need to index into `SparseSets` at all.